src/auto/inet_address_mask.rs

```diff
@@ -0,0 +1,183 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+use Error;
+use InetAddress;
+use SocketFamily;
+use ffi;
+use glib;
+use glib::Value;
+use glib::object::Downcast;
+use glib::object::IsA;
+use glib::signal::SignalHandlerId;
+use glib::signal::connect;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::boxed::Box as Box_;
+use std::fmt;
+use std::mem;
+use std::mem::transmute;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct InetAddressMask(Object<ffi::GInetAddressMask, ffi::GInetAddressMaskClass>);
+
+    match fn {
+        get_type => || ffi::g_inet_address_mask_get_type(),
+    }
+}
+
+impl InetAddressMask {
+    pub fn new(addr: &InetAddress, length: u32) -> Result<InetAddressMask, Error> {
+        unsafe {
+            let mut error = ptr::null_mut();
+            let ret = ffi::g_inet_address_mask_new(addr.to_glib_none().0, length, &mut error);
+            if error.is_null() { Ok(from_glib_full(ret)) } else { Err(from_glib_full(error)) }
+        }
+    }
+
+    pub fn new_from_string(mask_string: &str) -> Result<InetAddressMask, Error> {
+        unsafe {
+            let mut error = ptr::null_mut();
+            let ret = ffi::g_inet_address_mask_new_from_string(mask_string.to_glib_none().0, &mut error);
+            if error.is_null() { Ok(from_glib_full(ret)) } else { Err(from_glib_full(error)) }
+        }
+    }
+}
+
+impl PartialEq for InetAddressMask {
+    #[inline]
+    fn eq(&self, other: &Self) -> bool {
+        InetAddressMaskExt::equal(self, other)
+    }
+}
+
+impl Eq for InetAddressMask {}
+
+impl fmt::Display for InetAddressMask {
+    #[inline]
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", InetAddressMaskExt::to_string(self))
+    }
+}
+
+pub trait InetAddressMaskExt {
+    fn equal(&self, mask2: &InetAddressMask) -> bool;
+
+    fn get_address(&self) -> Option<InetAddress>;
```

Comment:
Can this really return `None`?

Reply:
I suppose it [cannot](https://github.com/GNOME/glib/blob/2dce3ce125cd0a58d4f49517dcd451857d62bdbd/gio/ginetaddressmask.c#L384).